### ISSUE TITLE
`docker swarm init` lists incorrect `join` command when running in PowerShell on Windows 2016

### ISSUE THREAD

**Issue by hairyhenderson** [moby/moby#32725]
**Description**

After doing a `docker swarm init` command in Windows 2016, in PowerShell, it outputs the following:

```console
Swarm initialized: current node (ugjxto86wtrj9hqaq65l4s7v2) is now a manager.

To add a worker to this swarm, run the following command:

    docker swarm join \
    --token SWMTKN-1-10mmx7lrwqgv70s47cdn1fpxjc3of0kll01fuude73773k6gpq-b4i96ydb9uys6bf714q7sadar \
    10.0.0.81:2377
```

The problem is, the `\` character isn't right in PowerShell, and the following error occurs:

```
PS C:\Windows\system32>  docker swarm join \
>>     --token SWMTKN-1-10mmx7lrwqgv70s47cdn1fpxjc3of0kll01fuude73773k6gpq-b4i96ydb9uys6bf714q7sadar \
>>     10.0.0.81:2377
At line:2 char:7
+     --token SWMTKN-1-10mmx7lrwqgv70s47cdn1fpxjc3of0kll01fuude73773k6g ...
+       ~
Missing expression after unary operator '--'.
At line:2 char:7
+     --token SWMTKN-1-10mmx7lrwqgv70s47cdn1fpxjc3of0kll01fuude73773k6g ...
+       ~~~~~
Unexpected token 'token' in expression or statement.
At line:3 char:10
+     10.0.0.81:2377
+          ~
Missing property name after reference operator.
    + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : MissingExpressionAfterOperator
```

**Steps to reproduce the issue:**
1. run `docker swarm init` on Windows 2016 in PowerShell
2. copy the `docker swarm join` command to a second Win2016 server and run it

**Describe the results you received:**

Encountered the error above.

**Describe the results you expected:**

I expected the node to join the swarm correctly 😉 

**Additional information you deem important (e.g. issue happens only occasionally):**

According to @mlaventure the correct character is a back-tick (`)

**Output of `docker version`:**

```
PS C:\Windows\system32> docker version
Client:
 Version:      17.05.0-dev
 API version:  1.29
 Go version:   go1.7.5
 Git commit:   a99c3bc
 Built:        Fri Mar 17 19:25:45 2017
 OS/Arch:      windows/amd64

Server:
 Version:      17.05.0-dev
 API version:  1.29 (minimum version 1.24)
 Go version:   go1.7.5
 Git commit:   a99c3bc
 Built:        Fri Mar 17 19:25:45 2017
 OS/Arch:      windows/amd64
 Experimental: false
```

**Output of `docker info`:**

```
PS C:\Windows\system32> docker version
Client:
 Version:      17.05.0-dev
 API version:  1.29
 Go version:   go1.7.5
 Git commit:   a99c3bc
 Built:        Fri Mar 17 19:25:45 2017
 OS/Arch:      windows/amd64

Server:
 Version:      17.05.0-dev
 API version:  1.29 (minimum version 1.24)
 Go version:   go1.7.5
 Git commit:   a99c3bc
 Built:        Fri Mar 17 19:25:45 2017
 OS/Arch:      windows/amd64
 Experimental: false
PS C:\Windows\system32> docker info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 16
Server Version: 17.05.0-dev
Storage Driver: windowsfilter
 Windows:
Logging Driver: json-file
Plugins:
 Volume: local
 Network: l2bridge l2tunnel nat null overlay transparent
Swarm: active
 NodeID: qeiwv102dqxfy8ulpwn47d8aw
 Is Manager: false
 Node Address: 10.0.0.20
 Manager Addresses:
  10.0.0.81:2377
Default Isolation: process
Kernel Version: 10.0 14393 (14393.729.amd64fre.rs1_release_inmarket_rim.170123-1753)
Operating System: Windows Server 2016 Datacenter
OSType: windows
Architecture: x86_64
CPUs: 2
Total Memory: 7GiB
Name: node1-9d4
ID: KPIL:4D6N:RYJP:XI4D:PBEA:DEQC:JSQ7:MMP6:KJOP:EI34:KJ3J:QTM2
Docker Root Dir: C:\ProgramData\docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):** Azure...

**Comment by thaJeztah** [comment]
/cc @jhowardmsft @johnstep @dnephin

thanks for reporting!

**Comment by jhowardmsft** [comment]
#dibs

**Comment by hairyhenderson** [comment]
@jhowardmsft I was planning on hacking on a PR tonight if you don't mind - looks like it's just in https://github.com/moby/moby/blob/99aaf6b9233d0a0a577e79f536288bca36794c50/cli/command/swarm/join_token.go#L96 - you OK with that, or do you _really_ want dibs? 🙂

**Comment by jhowardmsft** [comment]
Oh go for it! Should be a very simple fix

**Comment by hairyhenderson** [comment]
Thanks @jhowardmsft 😉

**Comment by hairyhenderson** [comment]
> I think the right fix is to just output it all on one line on Windows so it would work on native shells as well as Cygwin

I just occurred to me that this isn't really a Windows-only problem... If we want to join a Windows worker to a Linux manager, the Linux manager would spit out `\`s and the paste would still fail in the worker's PowerShell. And never mind the case where it's an all-Linux cluster but someone's using PowerShell for Linux as their main shell 😲...

It seems the only _correct_ solution would be to output it all on one line on all OSes. The catch is that it would be less human-readable, but I suppose it's better than causing strange errors to users unfamiliar with these sorts of things!

Anyway, it makes the PR even simpler - incoming... :wink:

**Comment by friism** [comment]
> It seems the only correct solution would be to output it all on one line on all OSes. The catch is that it would be less human-readable, but I suppose it's better than causing strange errors to users unfamiliar with these sorts of things!

Agree!

**Comment by jhowardmsft** [comment]
Yup. +1. Totally agreed. One line.

**Comment by hairyhenderson** [comment]
Fixed in #32731 😁

**Comment by albers** [comment]
+1 for one line. This is a technical instruction, accuracy beats readability.
